File: src/app/Class/Subgenre.ts
import { Genre } from './Genre';

export class Subgenre {

    private id: number;
    private parentGenre: Genre;
    private childGenre: Genre;

    constructor(obj?: any) {
        this.parentGenre = (obj && obj.parentGenre) || null;
        this.childGenre = (obj && obj.childGenre) || null;
    }

    // Getters

    getId(): number {
        return this.id;
    }

    getParentGenre(): Genre {
        return this.parentGenre;
    }

    getChildGenre(): Genre {
        return this.childGenre;
    }

    // Setters

    setParentGenre(parentGenre: Genre) {
        this.parentGenre = parentGenre;
    }

    setChildGenre(childGenre: Genre) {
        this.childGenre = childGenre;
    }

}

File: src/main/java/musiccube/entities/Subgenre.java
package musiccube.entities;

import javax.persistence.*;

@Entity
public class Subgenre {

    @Id
    @GeneratedValue
    @Column
    private int id;

    @ManyToOne
    private Genre parentGenre;

    @ManyToOne
    private Genre childGenre;

    public Subgenre() {}

    public Subgenre(Genre parentGenre, Genre childGenre) {
        this.parentGenre = parentGenre;
        this.childGenre = childGenre;
    }

    public int getId() {
        return id;
    }

    public void setId(int id) {
        this.id = id;
    }

    public Genre getParentGenre() {
        return parentGenre;
    }

    public void setParentGenre(Genre parentGenre) {
        this.parentGenre = parentGenre;
    }

    public Genre getChildGenre() {
        return childGenre;
    }

    public void setChildGenre(Genre childGenre) {
        this.childGenre = childGenre;
    }
}
